pkg/api/types.go

```diff
@@ -1713,8 +1713,12 @@ type ServiceSpec struct {
 	// This field will be ignored if the cloud-provider does not support the feature.
 	LoadBalancerIP string `json:"loadBalancerIP,omitempty"`
 
-	// Required: Supports "ClientIP" and "None".  Used to maintain session affinity.
+	// Optional: Supports "ClientIP" and "None".  Used to maintain session affinity.
 	SessionAffinity ServiceAffinity `json:"sessionAffinity,omitempty"`
+
+	// Optional: Supports "LoadBalancer" type on GCE and AWS
+	// Once specified, it will create corresponding firewall rules for the loadbalancer
```

Comment:
The comment should be less implementation-specific.

Maybe

If specified and supported by the platform, this will restrict traffic through the cloud-provider load-balancer will be restricted to the specified client IPs.  This field will be ignored if the cloud-provider does not support the feature."